lib/tools/system-calls.js

```diff
@@ -114,6 +127,51 @@ systemCallMethods.getBinaryFromSdkRoot = async function getBinaryFromSdkRoot (bi
   return binaryLoc;
 };
 
+/**
+ *  Returns the Android binaries locations
+ *
+ * @param {string} sdkRoot The path to Andoird SDK root.
+ * @param {string} fullBinaryName The name of full binary name.
+ */
+function getAndroidBinariesLocations (sdkRoot, fullBinaryName) {
+  return ANDROID_BINARIES_CANDIDATES.map((x) =>
+    path.resolve(sdkRoot, ...(_.isArray(x) ? x : [x]), fullBinaryName));
+}
+
+/**
+ * Retrieve full path to the given binary.
+ * This method does not have cache.
+ *
+ * @param {string} binaryName - Simple name of a binary file.
+ * @return {string} Full path to the given binary. The method tries
+ *                  to enumerate all the known locations where the binary
+ *                  might be located and stops the search as soon as the first
+ *                  match is found on the local file system.
+ * @throws {Error} If the binary with given name is not present at any
+ *                 of known locations or Android SDK is not installed on the
+ *                 local file system.
+ */
+async function getAndroidBinaryPath (binaryName) {
+  const fullBinaryName = getBinaryNameForOS(binaryName);
+  const sdkRoot = getSdkRootFromEnv();
+  const binaryLocs = getAndroidBinariesLocations(sdkRoot, fullBinaryName);
+  // get subpaths for currently installed build tool directories
+
+  let binaryLoc = null;
```

Comment:
there is no need in this variable. We could return the path immediately after `exists` is true